### PR TITLE
feat: add byInvitedPhoneNumbers to RetrieveThreadDTO.kt

### PR DESCRIPTION
**Pull request by DikenMaharjan**

### CODE REVIEW

File: src/main/kotlin/com/safehill/kclient/network/remote/RemoteServer.kt [hunk 1]
```diff
@@ -496,18 +481,18 @@ class RemoteServer(
...
-                        it.versionName == encryptedVersion.quality.value
+            asset.encryptedVersions.values.filter { filterVersions.contains(it.quality) }
+                .mapNotNull { encryptedVersion ->
+                    try {
+                        val serverAssetVersion = serverAsset.versions.first {
+                            it.versionName == encryptedVersion.quality.value
+                        }
+                        serverAssetVersion.presignedURL to encryptedVersion
+                    } catch (_: NoSuchElementException) {
+                        println("no server asset provided for version ${encryptedVersion.quality.value}")
```
Comment (gennaro-safehill) [inline src/main/kotlin/com/safehill/kclient/network/remote/RemoteServer.kt]: Can we avoid print statements?

**Review by gennaro-safehill — COMMENTED** [review]
Can we add support to retrieve all threads, by specifying no users and no phone numbers?

**Comment by DikenMaharjan** [comment]
> Can we add support to retrieve all threads, by specifying no users and no phone numbers?

We can retrieve all threads by calling listThreads()

File: src/main/kotlin/com/safehill/kclient/network/remote/RemoteServer.kt (hunk 1 above)
Comment (DikenMaharjan) [reply src/main/kotlin/com/safehill/kclient/network/remote/RemoteServer.kt]: Refactored.

**Review by gennaro-safehill — APPROVED** [review]
LGTM

**Pull request merged**